src/runtime_src/core/common/api/xrt_hw_context.cpp

```diff
@@ -127,13 +149,17 @@ namespace xrt {
 hw_context::
 hw_context(const xrt::device& device, const xrt::uuid& xclbin_id, const xrt::hw_context::cfg_param_type& cfg_param)
   : detail::pimpl<hw_context_impl>(std::make_shared<hw_context_impl>(device.get_handle(), xclbin_id, cfg_param))
-{}
+{
+  xrt_core::xdp::update_device(get_handle().get());
```

Comment:
I am little confused here.  Do we even need shared_from_this, why isn't update_device called with the shared pointer if you need to participate in ownership.  The raw pointer passed here is converted to a shared ptr immediately or later?  Not clear to me.

Reply:
Added explanation comment